### ISSUE TITLE
SC3045 false positive on ash (In dash, read -t is not supported)

### ISSUE THREAD

**Issue by felipecrs** [koalaman/shellcheck#3099]
#### For bugs
- Rule Id: SC3045
- My shellcheck version: 0.10.0
- [x] The rule's wiki page does not already cover this (https://shellcheck.net/wiki/SC3045)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
# shellcheck shell=ash

read -r -t 1 -n 1 keypress

echo "$keypress"
```

#### Here's what shellcheck currently says:

```
read -r -t 1 -n 1 keypress
        ^-- [SC3045](https://www.shellcheck.net/wiki/SC3045) (error): In dash, read -t is not supported.
```

#### Here's what I wanted or expected to see:

`read -t` is supported by both ASH and Busybox, I can confirm with:

```console
$ docker run --rm -it alpine ash -c 'read -r -t 1 -n 1 keypress && printf "\nKeypress: %s\n" "$keypress"'
s
Keypress: s
$ echo $?
0

$ docker run --rm -it alpine ash -c 'read -r -t 1 -n 1 keypress && printf "\nKeypress: %s\n" "$keypress"'
$ echo $?
1
```

```console
$ docker run --rm -it busybox sh -c 'read -r -t 1 -n 1 keypress && printf "\nKeypress: %s\n" "$keypress"'
s
Keypress: s
$ echo $?
0

$ docker run --rm -it busybox sh -c 'read -r -t 1 -n 1 keypress && printf "\nKeypress: %s\n" "$keypress"'
$ echo $?
1
```
